File: chapt2/chapt2_ws/src/demo_python_pkg/demo_python_pkg/person_node.py
import rclpy
from rclpy.node import Node

class PersonNode(Node):
    def __init__(self, node_name: str , name: str = "张三", age: int = 18) -> None:
        super().__init__(node_name)
        self.name = name
        self.age = age

    def eat(self, food_name: str = "小炒肉"):
        # print(f"{self.age}岁的{self.name}正在吃{food_name}")
        self.get_logger().info(f"{self.age}岁的{self.name}正在吃{food_name}")

def main():
    rclpy.init()
    node = PersonNode('zhangsan','张三', 18)
    node.eat('小炒肉')
    rclpy.spin(node)
    rclpy.shutdown()

File: chapt2/chapt2_ws/src/demo_cpp_pkg/src/person_node.cpp
#include <rclcpp/rclcpp.hpp>

class PersonNode : public rclcpp::Node
{
public:
    PersonNode(const std::string& node_name, const std::string& name, int age) : Node(node_name), name_(name), age_(age)
    {
        RCLCPP_INFO(this->get_logger(), "Person node has been started.");
    }

    void eat(const std::string &food)
    {
        RCLCPP_INFO(this->get_logger(), "I'm eating %s. My name is %s and I am %d years old.", food.c_str(), name_.c_str(), age_);
    }

private:
    std::string name_;
    int age_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<PersonNode>("person_node", "Jane", 18);
    node->eat("Apple");
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}